packages/store/ts/library/config/index.ts

```diff
@@ -227,20 +263,28 @@ export type MUDUserConfig<
     codegenDirectory?: string;
   };
 
-export type MUDConfig<
-  EnumNames extends StringForUnion = StringForUnion,
-  StaticUserTypes extends ExtractUserTypes<EnumNames> = ExtractUserTypes<EnumNames>
-> = MUDCoreConfig &
-  FullEnumsConfig<EnumNames> & {
-    tables: FullTablesConfig<StaticUserTypes, StaticUserTypes>;
-  };
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ExpandMUDUserConfig<T extends MUDCoreUserConfig>
+  extends OrDefaults<
+    T,
+    {
+      enums: Record<string, never>;
+      namespace: "";
+      storeImportPath: "@latticexyz/store/src/";
+      userTypesPath: "Types";
+      codegenDirectory: "codegen";
```

Comment:
I think it would be good to store these in some defaults const, so we can refer to them from the `zod` config expansion logic and have a single source of truth

Reply:
improved defaults for both `store` and `world` (took much of the code from your PR)